### PR TITLE
fix(oauth2) hash cache key

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Currently `oauth2` plugin does seem to use `access token` in a `cache_key` in plain. This is potentially dangerous and may be leaked through it to different places:
- memory
- admin api
  - logs

This commit fixes that by calling `sha1_bin` on that when generating a cache key.

### CODE REVIEW

File: kong/plugins/oauth2/access.lua [hunk 1]
```diff
@@ -605,7 +607,8 @@ local function retrieve_token(conf, access_token)
   local token, err
 
   if access_token then
-    local token_cache_key = kong.db.oauth2_tokens:cache_key(access_token)
+    local key = to_hex(sha1_bin(access_token))
+    local token_cache_key = kong.db.oauth2_tokens:cache_key(key)
```
Comment (p0pr0ck5) [inline kong/plugins/oauth2/access.lua]: does this properly handle cache invalidations?

Comment (bungle) [reply kong/plugins/oauth2/access.lua]: Good question. I don't think so.  Any solutions?

Comment (bungle) [reply kong/plugins/oauth2/access.lua]: I think tokens are only inserted, and deleted, and selected. So perhaps it is fine for now?

Comment (bungle) [reply kong/plugins/oauth2/access.lua]: No it is not fine, I will write an invalidation handler.

**Comment by hishamhm** [comment]
Does Kong ever outputs raw `cache_key` values in the logs in a level other than `DEBUG`?

**Comment by hishamhm** [comment]
I think always enabling read-before-write for `update` (782f05d)  is very controversial and needs some good discussion and consideration. I'm pointing this out so a major core change doesn't go "unnoticed" along with a plugin fix.

**Comment by bungle** [comment]
> I think always enabling read-before-write for `update` ([782f05d](https://github.com/Kong/kong/commit/782f05d88c1fe4c268129b435ebb0db8984376d8)) is very controversial and needs some good discussion and consideration.

Yes, I will move this to waiting. The thing is, I think we broke `data.old_entity` with `1.0`. And that also warrants discussion.

**Pull request closed without merging**